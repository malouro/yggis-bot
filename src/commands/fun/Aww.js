import RedditCommand from './Reddit';
import Command from '../../classes/Command';

export default class Aww extends Command {
	constructor({ t } = { t: () => null }) {
		super({
			name: 'Aww',
			category: 'fun',
			description: 'Grab a cute & cuddly post from the /r/aww subreddit.',
		});

		this.t = t;
		this.translationKey = 'reddit';
		this.translationKeyForDescriptions = 'aww';

		this.name =
			this.t('COMMANDS', this.translationKeyForDescriptions, 'name') ||
			this.name;
		this.description =
			this.t('COMMANDS', this.translationKeyForDescriptions, 'description') ||
			this.description;

		this.subreddit = 'aww';
	}

	action({ message, logger }) {
		const fireRedditAction = RedditCommand.prototype.action.bind(this, {
			message,
			logger,
		});

		fireRedditAction();
	}
}
