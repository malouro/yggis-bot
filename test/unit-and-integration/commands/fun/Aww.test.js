import Aww from '../../../../src/commands/fun/Aww';

import { runCommand, MockTranslateFunc } from '../../../testHelpers';

jest.mock('feed-read-parser', () => (url, callback) => {
	switch (url) {
		// Modified result from fetching from r/aww .rss
		case 'https://www.reddit.com/r/aww/.rss':
			return callback(null, [
				{
					title: 'POST_TITLE',
					// Imitate XML that can return from feed-read-parser
					content: `<table> <tr><td> <a href="https://www.reddit.com/r/aww/comments/SOME_POST/">
			<img src="https://b.thumbs.redditmedia.com/8nS_rT2NxKhqMkxGCgO8_bLhYYRkqnKz_9hp2lHvP8Q.jpg"
			alt="🐱" title="🐱" /> </a> </td><td> &#32; submitted by &#32;
			<a href="https://www.reddit.com/user/SOME_AUTHOR"> /u/SOME_AUTHOR </a>
			<span><a href="https://i.redd.it/pqdffk5lsp751.jpg">[link]</a></span> &#32; <span>
			<a href="https://www.reddit.com/r/aww/comments/SOME_POST/">[comments]</a></span> </td></tr></table>`,
					published: 'SOME_DATE',
					author: '/u/SOME_AUTHOR',
					link: 'https://www.reddit.com/r/aww/comments/SOME_POST/',
					feed: {
						source: 'https://www.reddit.com/r/aww/.rss',
						link: 'https://www.reddit.com/r/aww/',
						name: 'aww',
					},
				},
			]);

		default:
			break;
	}
});

describe('Aww Command', () => {
	const AwwCommand = new Aww({ t: MockTranslateFunc() });

	test('returns a post from r/aww', async () => {
		const result = await runCommand(AwwCommand, { args: ['!aww']});

		expect(result.message.channel.send.mock.results[0].value).toMatchSnapshot();
	});
});
