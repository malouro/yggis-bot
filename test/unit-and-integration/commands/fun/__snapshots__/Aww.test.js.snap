// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Aww Command returns a post from r/aww 1`] = `
"**POST_TITLE**
posted by /u/SOME_AUTHOR
https://i.redd.it/pqdffk5lsp751.jpg
<https://www.reddit.com/r/aww/comments/SOME_POST/>"
`;
